app/services/finance/stripe_payment_intent_update_service.rb

```diff
@@ -32,6 +32,7 @@ def payment_intent_id
 
   def create_payment_transaction
     attributes = {
+      account: invoice.buyer_account,
```

Comment:
PaymentTransation belongs to "account", but
Invoice belongs to "buyer_account" and "provider_account".

So, could it ever be `account: invoice.provider_account`? Do you know why?

Reply:
yeah, I wondered the same and I wasn't 100% sure but I finally assumed it will always be `invoice.buyer_account` because:

1. An invoice will always have somebody who pays and somebody who gets paid.
2. The expected account here is the one paying the invoice.
3. AFAIK, only two situations can happen:
   1. A buyer paying a provider
   2. A provider paying master (AFAIK, this is only used in SaaS so tenants pay us)
4. In the second scenario the provider would be in fact `invoice.buyer_account` and master would be `invoice.provider_account`. So in both scenarios the one paying is `buyer_account`.
5. Finally, the second scenario cannot happen anyway, because master only accepts payments through Braintree and this service is for Stripe.